rpc/api/miner.go

```diff
@@ -126,7 +126,7 @@ func (self *minerApi) SetExtra(req *shared.Request) (interface{}, error) {
 		return nil, err
 	}
 
-	if uint64(len(args.Data)) > params.MaximumExtraDataSize.Uint64()*2 {
+	if uint64(len(args.Data)) > params.MaximumExtraDataSize.Uint64() {
```

Comment:
You are correct stating that that it doesn't always require hex, however, the RPC (which is used by the console) and Web3 should convert that string to a hex value :-)